R/geom-errorbar.R

```diff
@@ -74,3 +102,9 @@ GeomErrorbar <- ggproto("GeomErrorbar", Geom,
 
   rename_size = TRUE
 )
+
+#' @rdname ggplot2-ggproto
+#' @format NULL
+#' @usage NULL
+#' @export
+GeomErrorbarh <- ggproto("GeomErrorbarh", GeomErrorbar)
```

Comment:
Couldn't we throw in a deprecation warning in `setup_param()` and see how that works?

If we give it the same ID as the one in the constructor we can ensure it only fires once for those that uses the constructor

Reply:
Yeah works great:

``` r
devtools::load_all("~/packages/ggplot2")
#> ℹ Loading ggplot2

df <- data.frame(
  trt = factor(c(1, 1, 2, 2)),
  resp = c(1, 5, 3, 4),
  group = factor(c(1, 2, 1, 2)),
  upper = c(1.1, 5.3, 3.3, 4.2),
  lower = c(0.8, 4.6, 2.4, 3.6)
)

ggplot(df, aes(resp, trt, colour = group)) +
  layer(
    geom = "errorbarh",
    stat = "identity",
    position = "identity",
    mapping = aes(xmin = lower, xmax = upper),
    params = list(width = 0.2, na.rm = FALSE, orientation = NA)
  )
#> Warning: `geom_errobarh()` was deprecated in ggplot2 3.5.2.
#> ℹ Please use the `orientation` argument of `geom_errorbar()` instead.
#> This warning is displayed once every 8 hours.
#> Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
#> generated.
```

![](https://i.imgur.com/kC2R3Pa.png)<!-- -->

``` r

ggplot(df, aes(resp, trt, colour = group)) +
  geom_errorbarh(aes(xmin = lower, xmax = upper))
```

![](https://i.imgur.com/TarUCAL.png)<!-- -->

<sup>Created on 2024-09-13 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>